src/dynamic_boundary_conditions/tide/sea_level_rise_data.py

```diff
@@ -3,170 +3,118 @@
 
-    Parameters
-    ----------
-    slr_data_dir : pathlib.Path
-        The directory containing the downloaded sea level rise data files.
+def modify_slr_data_from_takiwa() -> gpd.GeoDataFrame:
+    """
+    Modify sea level rise data stored under dictionary to a GeoDataFrame and return.
 
     Returns
     -------
     gpd.GeoDataFrame
         A GeoDataFrame containing the sea level rise data from the NZ Sea level rise datasets.
-
-    Raises
-    ------
-    FileNotFoundError
-        If the sea level rise data directory does not exist or if there are no CSV files in the specified directory.
     """
-    # Check if the sea level rise data directory exists
-    if not slr_data_dir.exists():
-        raise FileNotFoundError(f"'sea_level_rise' data directory not found: '{slr_data_dir}'.")
-    # Check if there are any CSV files in the specified directory
-    if not any(slr_data_dir.glob("*.csv")):
-        raise FileNotFoundError(f"'sea_level_rise' data files not found in: '{slr_data_dir}'")
-    # Create an empty list to store the sea level rise datasets
-    slr_nz_list = []
-    # Loop through each CSV file in the specified directory
-    for file_path in slr_data_dir.glob("*.csv"):
-        # Read the CSV file into a pandas DataFrame using pyarrow
-        slr_region = csv.read_csv(file_path).to_pandas()
-        # Extract the region name from the file name and add it as a new column in the DataFrame
-        file_name = file_path.stem
-        start_index = file_name.find('projections_') + len('projections_')
-        end_index = file_name.find('_region')
-        region_name = file_name[start_index:end_index]
-        slr_region['region'] = region_name
-        # Append the DataFrame to the list
-        slr_nz_list.append(slr_region)
-        # Log that the file has been successfully loaded
-        log.info(f"Successfully loaded the '{file_path.name}' data file.")
-    # Concatenate all the dataframes in the list and add geometry column
-    slr_nz = pd.concat(slr_nz_list, axis=0).reset_index(drop=True)
-    geometry = gpd.points_from_xy(slr_nz['lon'], slr_nz['lat'], crs=4326)
-    slr_nz_with_geom = gpd.GeoDataFrame(slr_nz, geometry=geometry)
-    # Convert all column names to lowercase
-    slr_nz_with_geom.columns = slr_nz_with_geom.columns.str.lower()
+    # Generate a dictionary to store the sea level rise dataset
+    slr_nz_dict = get_slr_data_from_takiwa()
+
+    # Create a copy dataframe for NZ_VLM_final_May24
+    slr_nz = slr_nz_dict["NZ_VLM_final_May24"].copy(deep=True)
+    # Merge Site Details dataframe and Sea level projections tables WITH and WITHOUT VLM
+    slr_nz_merge_list = []
+    for vlm_name in ["NZSeaRise_proj_vlm", "NZSeaRise_proj_novlm"]:
+        slr_nz_merge = slr_nz.merge(
+            slr_nz_dict[vlm_name],
+            left_on='Site ID', right_on='site',
+            how='left', validate='1:m',
+            suffixes=('_y', '')
+        )
```

Comment:
```suggestion
        slr_nz_merge = slr_nz.merge(
            slr_nz_dict[vlm_name],
            left_on='Site ID', 
            right_on='site',
            how='left'
        )
```

Reply:
I'd like to see if the data is any different after this. I'm not expecting any changes.

Reply:
> I'd like to see if the data is any different after this. I'm not expecting any changes.

I have tested this myself and found no differences in the data. However, to ensure thoroughness, @Martin20494, could you please verify this again?